backend/eventloop/shared_resource_loop/sharedresourceloop_repositorycredential.go

```diff
@@ -0,0 +1,663 @@
+	if err != nil {
+		errV := fmt.Errorf("unable to create operation: %v", err)
+		return errV, ""
+	}
+
+	l.Info("operation has been created", "CR", operationCR, "DB", operationDB)
+
+	return nil, operationDB.Operation_id
+}
+
+// Updates the given repository credential CR's status condition to match the given condition and additional checks.
+// If there is an existing status condition with the exact same status, reason and message, no update is made in order
+// to preserve the LastTransitionTime (see https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#Condition.LastTransitionTime )
+func UpdateGitopsDeploymentRepositoryCredentialStatus(ctx context.Context, repositoryCredential *managedgitopsv1alpha1.GitOpsDeploymentRepositoryCredential, client client.Client, secret *corev1.Secret, log logr.Logger) error {
+
+	// if the condition was sent along with the function call, we don't need to perform additional checks
+	newConditions := checkForValidRepositoryCredential(repositoryCredential, ctx, secret)
+
+	needToUpdateConditions := false
+	for _, condition := range newConditions {
+		// do nothing if appset already has same condition
+		for _, c := range repositoryCredential.Status.Conditions {
+			if c.Type == condition.Type && (c.Reason != condition.Reason || c.Status != condition.Status || c.Message != condition.Message) {
+				needToUpdateConditions = true
+				break
+			}
+		}
+	}
+	evaluatedTypes := map[string]bool{
+		managedgitopsv1alpha1.GitOpsDeploymentRepositoryCredentialConditionErrorOccurred:             true,
+		managedgitopsv1alpha1.GitOpsDeploymentRepositoryCredentialConditionValidRepositoryUrl:        true,
+		managedgitopsv1alpha1.GitOpsDeploymentRepositoryCredentialConditionValidRepositoryCredential: true,
+	}
+
+	if needToUpdateConditions || len(repositoryCredential.Status.Conditions) < 3 {
+
+		// 1) Attempt to get the latest gitopsDeploymentRepositoryCredentialCR from the namespace
+		if err := client.Get(ctx, types.NamespacedName{Namespace: repositoryCredential.Namespace, Name: repositoryCredential.Name},
+			repositoryCredential); err != nil {
+
+			if apierr.IsNotFound(err) {
+				repositoryCredential = nil
+
+			} else {
+				// Something went wrong, retry
+				vErr := fmt.Errorf("unexpected error in retrieving repository credentials: %v", err)
+				log.Error(err, vErr.Error(), "DebugErr", errGenericCR, "CR Name", repositoryCredential, "Namespace", repositoryCredential.Namespace)
+
+				return nil
+
+			}
+		}
+		repositoryCredential.Status.SetConditions(
+			newConditions, evaluatedTypes,
+		)
+		// Update the GitOpsDeploymentRepositoryCredential CR
+		if err := client.Status().Update(ctx, repositoryCredential); err != nil {
+			log.Error(err, "updating repository credential CR's status condition")
+		}
+	}
+	return nil
+
+}
+
+func checkForValidRepositoryCredential(repositoryCredential *managedgitopsv1alpha1.GitOpsDeploymentRepositoryCredential, ctx context.Context, secret *corev1.Secret) []*metav1.Condition {
```

Comment:
`checkForValidRepositoryCredential` is currently missing full test coverage for all of the if-else branches: you can check this by running `make test` in the `backend/` folder, and then doing `Go: Apply Cover Profile', and passing the path to the cover.out file at `backend/cover.out`.